### ISSUE TITLE
Comments: Clear AWS Credentials Cache

### ISSUE THREAD

**Issue by jayair** [AnomalyInnovations/serverless-stack-com#50]

**Comment by d3sandoval** [comment]
Hi there, per my comment in https://github.com/AnomalyInnovations/serverless-stack-com/issues/55#issuecomment-318859903 I am still able to see the same "notes list" across users.

Here's an example of me creating a note in one user and seeing it in the other:
![](https://media.giphy.com/media/l41JF8XUFuiwrRLQA/giphy.gif)

I have followed the instructions in the chapter exactly and am still encountering this issue... could there be a problem in code _outside_ of app.js?

My package.json states that I am on aws-sdk version "^2.88.0"

**Comment by jayair** [comment]
@d3sandoval Do you still see the old user's notes after you refresh the page or is it only happening right after you logout and login?

**Comment by d3sandoval** [comment]
@jayair no. refreshing fixes the issue. Is there an elegant way to get this functionality to work without refreshing? If not, do you know of a good place in the app to force the user to refresh?

**Comment by jayair** [comment]
@d3sandoval I think we broke it with the latest update. The simplest fix right now would be to change this line https://github.com/AnomalyInnovations/serverless-stack-demo-client/blob/master/src/App.js#L87

```js
AWS.config.credentials.clearCachedId();
```

to this

```js
delete AWS.config.credentials;
```

I'll need to test it a bit more and update the tutorial.

**Comment by d3sandoval** [comment]
@jayair that's worse :/ I get an error message after the login loop... refreshing still fixes it:
![](https://media.giphy.com/media/26BnbClkkduI1f4wo/giphy.gif)
For now, I'll just add a refresh on logout.

**Comment by jayair** [comment]
@d3sandoval I see. Yeah I'll have to take a look and put in a better fix then.

**Comment by jayair** [comment]
@d3sandoval I haven't had a chance to update the tutorial yet but if you are still looking for a fix. Try replacing this line - https://github.com/AnomalyInnovations/serverless-stack-demo-client/blob/master/src/App.js#L87

with this.

```js
AWS.config.credentials.clearCachedId();
AWS.config.credentials = new AWS.CognitoIdentityCredentials({ });
```

**Comment by limifont** [comment]
Adding the new AWS.CognitoIdentityCredentials({}) seems to have fixed the issue!

**Comment by gregt590** [comment]
A related problem occurs if you first attempt to login to a user that doesn't exist and get an 
UserNotFound Exception, and then login to a valid user and attempt to create a new note with an attachment which will generate an AccessDenied error, because it was using the wrong access credentials which hadn't expired yet and didn't have a valid identityId defined which also causes it to attempt to create the S3 file: "undefined-1504037928967-filename.ext".

Resetting the AWS.config.credentials as described above fixed this issue as well.

**Comment by jayair** [comment]
@gregt590 Ah thats true. The update with resetting the credentials is coming out shortly.

**Comment by QuantumInformation** [comment]
Can we close this?

**Comment by jayair** [comment]
@QuantumInformation You mean this issue? It's the comments thread for this chapter.

**Comment by QuantumInformation** [comment]
oops, sorry )
